markdown.go

```diff
@@ -639,9 +639,6 @@ func scanLinkRef(p *parser, data []byte, i int) (linkOffset, linkEnd, titleOffse
 	for i < len(data) && data[i] != ' ' && data[i] != '\t' && data[i] != '\n' && data[i] != '\r' {
 		i++
 	}
-	if i == len(data) {
-		return
-	}
```

Comment:
Ohh, that's a very good question that I don't have a good answer to. It's only being called from `isReference`, which in turn is only called from the `paragraph` block parser, but that doesn't say much about preconditions. Other than that, it will take some time to study those nearly 200 lines carefully enough to tease out all preconditions. I'll take a note to do that when I have chance.